core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -289,10 +300,11 @@ class Partition(val topicPartition: TopicPartition,
                 delayedOperations: DelayedOperations,
                 metadataCache: MetadataCache,
                 logManager: LogManager,
-                alterIsrManager: AlterPartitionManager) extends Logging {
+                alterIsrManager: AlterPartitionManager,
+                @volatile private var _topicId: Option[Uuid] = None // TODO: merge topicPartition and _topicId into TopicIdPartition once TopicId persist in most of the code
```

Comment:
Can we add jira link to the comment? The reader can trace the updates easily.

Reply:
there is a jira that will be address this already which is [KAFKA-16212](https://issues.apache.org/jira/browse/KAFKA-16212) I'll update the comment